src/pudl/settings.py

```diff
@@ -243,41 +244,84 @@ class Eia860Settings(GenericDatasetSettings):
         data_source: DataSource metadata object
         years: list of years to validate.
         eia860m: whether or not to incorporate an EIA-860m month.
-        eia860m_year_month ClassVar[str]: The 860m year-month to incorporate.
+        eia860m_year_months ClassVar[str]: The 860m year-months to incorporate.
     """
 
     data_source: ClassVar[DataSource] = DataSource.from_id("eia860")
     years: list[int] = data_source.working_partitions["years"]
-
     eia860m: bool = True
-    eia860m_year_month: ClassVar[str] = max(
-        DataSource.from_id("eia860m").working_partitions["year_months"]
-    )
+    all_eia860m_year_months: list[str] = DataSource.from_id(
+        "eia860m"
+    ).working_partitions["year_months"]
+    eia860m_year_months: list[str] = Field(validate_default=True, default=[])
 
-    @field_validator("eia860m")
+    @field_validator("eia860m_year_months")
     @classmethod
-    def check_eia860m_year_month(cls, eia860m: bool) -> bool:
-        """Check 860m date-year is exactly one year after most recent working 860 year.
-
-        Args:
-            eia860m: True if 860m is requested.
+    def add_other_860m_years(cls, v, info: ValidationInfo) -> list[str]:
+        """Find extra years from EIA860m if applicable.
 
-        Returns:
-            eia860m: True if 860m is requested.
+        There's a gap in reporting (after the new year but before the EIA 860 early
+        release data in June) when we rely on two years worth of EIA860m data instead
+        of just one. This function adds the last available month_year values for each
+        year of 860m data that is not yet available in 860.
 
-        Raises:
-            ValueError: the 860m date is within 860 working years.
         """
-        eia860m_year = pd.to_datetime(cls.eia860m_year_month).year
-        expected_year = max(cls.data_source.working_partitions["years"]) + 1
-        if eia860m and (eia860m_year != expected_year):
-            raise AssertionError(
-                f"Attempting to integrate an eia860m year "
-                f"({eia860m_year}) from {cls.eia860m_year_month} not immediately following"
-                f"the eia860 years: {cls.data_source.working_partitions['years']}. "
-                f"Consider switching eia860m parameter to False."
+        extra_eia860m_year_months_string = []
+        if info.data["eia860m"]:
+            all_eia860m_years = set(
+                pd.to_datetime(info.data["all_eia860m_year_months"]).year
             )
-        return eia860m
+            # The years in 860m that are not in 860
+            extra_eia860m_years = {
+                year
+                for year in all_eia860m_years
+                if year not in DataSource.from_id("eia860").working_partitions["years"]
+            }
```

Comment:
I changed this from `info.data["years"]` so that it only populates the `extra_eia860m_years` list with years that aren't available in 860 (even if they aren't specified in `years`)